pkg/segment/writer/rawchecker.go

```diff
@@ -51,13 +51,17 @@ func ApplySearchToMatchFilterRawCsg(match *MatchFilter, col []byte) (bool, error
 	// todo MatchWords struct can store bytes
 	if match.MatchOperator == And {
 		var foundQword bool = true
-		if match.MatchType == MATCH_PHRASE {
-			foundQword = utils.IsSubWordPresent(col[idx:idx+clen], match.MatchPhrase)
+		if match.Regexp != nil {
```

Comment:
this check should happen in the `if match.MatchType == MATCH_PHRASE` as mentioned in the earlier comment:
when matchtype is match_phrase only then invoke the "new logic", and the new logic is  : if regexp is not nil then do regex match, else do the issubword match